"Show related card" option added to ingame context menu

Vafthrudnir opened the pull request:
## Related Ticket(s)
- Fixes #1993 

## Short roundup of the initial problem
There was no way to know the details of a card's reverse side without actually creating that card in-game.

## What will change with this Pull Request?
- "Show Card" option added to in-game right click menus in all zones and pop-up card lists (e.g. on top cards of library view)
- Each related card is a menu new menu item, when clicked it updates the card info widget with its details
- Context menu on cards without related cards haven't changed

## Screenshots
<img width="510" alt="screenshot 2018-02-17 02 43 38" src="https://user-images.githubusercontent.com/7460172/36339130-70585b52-138c-11e8-9bbd-014740f73ccb.png">
<img width="612" alt="screenshot 2018-02-17 02 43 24" src="https://user-images.githubusercontent.com/7460172/36339131-709cd700-138c-11e8-8969-8c65d1e04a62.png">
<img width="607" alt="screenshot 2018-02-17 02 43 17" src="https://user-images.githubusercontent.com/7460172/36339132-70eaff52-138c-11e8-8a77-4cb8bafd7178.png">
<img width="401" alt="screenshot 2018-02-17 02 43 09" src="https://user-images.githubusercontent.com/7460172/36339133-711f9cc6-138c-11e8-8a33-b0ba3bffea9f.png">

Daenyth commented:
I'm concerned that "Show card" might be misleading to users. It's very similar to "reveal card"

Vafthrudnir commented:
Yeah, that really may be so. Do you have a suggestion what to rename it to?

Daenyth commented:
"Display card info"? Maybe? @ZeldaZach @tooomm thoughts?

ZeldaZach commented on cockatrice/src/player.cpp:
```diff
@@ -2685,6 +2686,28 @@ void Player::updateCardMenu(const CardItem *card)
+    cardMenu->addSeparator();
+    auto *signalMapper = new QSignalMapper(this);
+    for (const CardRelation *relatedCard : relatedCards) {
+        QAction *viewCard = cardMenu->addAction("Show Card: \"" + relatedCard->getName() + "\"");
```
`...->addAction(tr("View Card") + ":\"" + relatedCard->getName() + "\"");`

Daenyth replied:
Note especially the `tr`, that's important so the string gets translated for other languages

ZeldaZach commented on cockatrice/src/player.cpp:
```diff
@@ -2685,6 +2686,28 @@ void Player::updateCardMenu(const CardItem *card)
             cardMenu->addAction(aClone);
         }
     }
+    addRelatedCardView(card, cardMenu);
```
While this location is good for cards in hand, it's really inconvenient for cards in play. Either we should make this feature only work in hand, or we should place it somewhere better when the card's in play. Similar issues on the stack.

<img width="327" alt="screenshot 2018-02-16 02 35 21" src="https://user-images.githubusercontent.com/7460172/36297685-27ae95b2-12c2-11e8-82e0-0cddcb2e3988.png">
<img width="358" alt="screenshot 2018-02-16 02 35 25" src="https://user-images.githubusercontent.com/7460172/36297686-27c95906-12c2-11e8-9a7b-4cc23d5afa58.png">
<img width="346" alt="screenshot 2018-02-16 02 35 51" src="https://user-images.githubusercontent.com/7460172/36297687-27d640da-12c2-11e8-89bc-3b9c498e56af.png">

Daenyth replied:
For cards in play I'd actually put it near the token menu, maybe above or below

ZeldaZach replied:
My thought was to `view related card images >` submenu it

Vafthrudnir commented:
I left the option at the end of the context menu, but changed its name ("View related card info:") and added it as a submenu. Also added the `tr()` call.

Daenyth commented on cockatrice/src/player.cpp:
```diff
@@ -2612,6 +2612,8 @@ void Player::updateCardMenu(const CardItem *card)
 
         if (card->getZone()) {
             if (card->getZone()->getName() == "table") {
+                // Card is on the battlefield
```
I'm not sure this comment adds any clarity; the line above says the same thing.

ZeldaZach replied:
Added since I added to all of them

Daenyth commented on cockatrice/src/player.cpp:
```diff
@@ -2657,22 +2660,27 @@ void Player::updateCardMenu(const CardItem *card)
                 cardMenu->addAction(aClone);
                 cardMenu->addMenu(moveMenu);
             } else {
+                // Card is in hand
```
in hand or any custom zone that the server is using which the client isn't aware of, would be more accurate.

ZeldaZach replied:
True, i'll make this change

ZeldaZach commented:
Screenshots Updated

ZeldaZach commented:
I see... Can you add it in for that case?

ZeldaZach commented:
I just have to say, with the tokens, it seems really weird to see the same card twice. Maybe we can move the create tokens to a sub dialog? What do y'all think?

Daenyth commented:
We probably shouldn't nest the menu too deeply, since that makes it more
annoying to use, and related cards are such a common interaction

On Sat, Feb 17, 2018 at 3:03 AM Zach H <notifications@github.com> wrote:

> I just have to say, with the tokens, it seems really weird to see the same
> card twice. Maybe we can move the create tokens to a sub dialog? What do
> y'all think?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/3115#issuecomment-366425099>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA5NCBCuZMvzVaGVf5hHGo2F8w05yvhks5tVoe2gaJpZM4SHmES>
> .
>
